crates/hir-def/src/body/lower.rs

```diff
@@ -199,6 +204,10 @@ impl ExprCollector<'_> {
         self.source_map.pat_map.insert(src, id);
         id
     }
+    // desugared pats don't have ptr, that's wrong and should be fixed somehow.
```

Comment:
This should be a FIXME, and ideally we should fix this before doing more desugaring as this will degrade IDE features otherwise

Reply:
Added the FIXME.

I copied this comment from the existing `alloc_expr_desugared`. I didn't use `alloc_expr_desugared` and used the syntax_ptr of the root expression for generated expressions. Is it correct? And is there any feature that might not work due usage of `alloc_pat_desugared` in try desugaring?

Reply:
So I took another look, this will only be problematic for diagnostics it seems (that is the desugared stuff causes some diagnostics to trigger, we won't show them then).